### PR TITLE
Twitch addon support

### PR DESCRIPTION
**Pull request by crutchcorn**
This PR is meant to be a precursor to a blog post PR that I'll be releasing soon.

Since that blog post has a Twitch clip within it, I wanted to embed it properly. This PR adds the ability to embed it

I'm requesting @evelynhathaway to review this, since it makes minor changes to how deploy-adjacent envs and variables are set.

I am 99% confident this shouldn't break anything, but figured I'd check with her first :)

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -4,11 +4,21 @@ let CONSTS = require("./config/gatsby-config-consts");
...
+		case "production":
+			siteUrl = "https://unicorn-utterances.com";
+			break;
+		case "local":
+			siteUrl = "localhost:8000";
+			break;
+		default:
+			siteUrl = "https://beta.unicorn-utterances.com";
+	}
+}
```
Comment (crutchcorn) [inline gatsby-config.js:21]: This should keep previous behavior, but is one change made to a deploy related variable

File: package.json [hunk 2]
```diff
@@ -20,9 +20,9 @@
   },
   "scripts": {
     "build": "gatsby build",
-    "build-beta": "BUILD_ENV=development npm run build",
+    "build-beta": "cross-env BUILD_ENV=development npm run build",
```
Comment (crutchcorn) [inline package.json:23]: Likewise, we've enabled `cross-env` here. This _should_ function as-intended, but just wanted to highlight

File: plugins/gatsby-remark-twitch/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        width='${options.width}'
+        frameborder='0'
+        scrolling='no'
+        allowfullscreen='true'>
+      </iframe></div>`;
+		}
+	});
+
+	return markdownAST;
+};
```
Comment (crutchcorn) [inline plugins/gatsby-remark-twitch/index.js:85]: This code is based on forked code.

We've made an upstream PR to fix the issues. When merged, our local copy can be deleted:

https://github.com/octahedroid/gatsby-remark-twitch/pull/3

**Comment by crutchcorn** [comment]
FWIW, this _seems_ to still be working for our deploy of dev:

https://twitch-addon.unicorn-utterances.com/robots.txt

**Comment by evelynhathaway** [comment]
Looks great. Thanks for taking this on.

**Pull request merged**